src/warning.js

```diff
@@ -0,0 +1,48 @@
+QUnit.warning = (function() {
+    var messages = {
+        asyncTest: "DEPRECATED: QUnit.asyncTest() will " +
+            "be removed in QUnit 2.0.",
+        "init": "DEPRECATED: QUnit.init() will throw " +
+            "an error in 2.0, and will be removed in 2.1.",
+        loggingCallbacks: "DEPRECATED: Logging callbacks being assigned " +
+            "with new values instead of called as functions " +
+            "will be removed on QUnit 2.0.0.",
+        "module-setup": "DEPRECATED: beforeEach were previously named " +
+            "setup, which still exist and will be removed in QUnit 2.0.0.",
+        "module-teardown": "DEPRECATED: afterEach were previously named " +
+            "teardown, which still exist and will be removed in QUnit 2.0.0.",
+        push: "DEPRECATED: QUnit.push() is deprecated and " +
+            "it's recommended to use it through its direct reference " +
+            "in the assertion context.",
+        reset: "DEPRECATED: Use multiple tests instead of resetting inside a test. " +
+            "Use testStart or testDone for custom cleanup. " +
+            "This method will throw an error in 2.0, and will be removed in 2.1.",
+        start: "DEPRECATED: The functionality of QUnit.start() will be altered. " +
+            "In QUnit 2.0, invoking it will ONLY affect the " +
+            "`QUnit.config.autostart` blocking behavior.",
+        stop: "DEPRECATED: QUnit.stop() will be removed in QUnit 2.0.",
+        "testDone-duration": "DEPRECATED: duration property will be removed " +
+            "in 2.0.0, use runtime instead.",
+        "exported-globals": "DEPRECATED: QUnit 2.0 will not export its API " +
+            "methods to the window object."
+    };
+
+    var once = [];
```

Comment:
yes, it makes sense. 

Having object values on this array would help with this problem too.

Even if we don't have too much extra information to store for now, ordinary objects would allow it to be more extensive.

I wish we could already use Maps and Sets here.